src/Core/Grid/Query/CustomerThreadQueryBuilder.php

```diff
@@ -0,0 +1,226 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Builds search and count query builders for customer thread grid.
+ */
+class CustomerThreadQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $searchCriteriaApplicator;
+
+    /**
+     * @var int[]
+     */
+    private $contextShopIds;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+     * @param array $contextShopIds
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
+        array $contextShopIds
+    ) {
+        parent::__construct($connection, $dbPrefix);
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+        $this->contextShopIds = $contextShopIds;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria): QueryBuilder
+    {
+        $qb = $this->getQueryBuilder($searchCriteria)
+            ->select('ct.*, CONCAT(c.`firstname`," ",c.`lastname`) as customer')
+            ->addSelect('cm.message, cm.private, cl.name as contact, l.name as langName')
+            ->addSelect('s.name as shopName')
+            ->addSelect('(
+				SELECT IFNULL(CONCAT(LEFT(e.`firstname`, 1),". ",e.`lastname`), "--")
+				FROM `' . _DB_PREFIX_ . 'customer_message` cm2
+				INNER JOIN ' . _DB_PREFIX_ . 'employee e
+					ON e.`id_employee` = cm2.`id_employee`
+				WHERE cm2.id_employee > 0
+					AND cm2.`id_customer_thread` = ct.`id_customer_thread`
+				ORDER BY cm2.`date_add` DESC LIMIT 1
+			) as employee')
```

Comment:
Do you really need to perform this subquery to get the employee's data? Since the join is already done in the base query maybe there's a simpler way? Like maybe:

```suggestion
            ->addSelect('IFNULL(CONCAT(LEFT(e.`firstname`, 1),". ",e.`lastname`), "--") as employee')
```

But I'm probably missing something that prevents doing this.

Reply:
Oh, maybe the purpose here is to get the employee attached to the latest customer message linked to the thread? If so can you add a small comment to explain this sub query please?

Reply:
You didn't answer my question here. Do you really need this subquery? From what I understood you need the employee associated to the last message from the thread right?

But you already perform a `leftJoin` to get `customer_message` and even `employee` table, so what's the use of this extra subquery? If you want to be sure that the joined table is a message from an Employee you can adapt you join condition:

```php
            ->leftJoin(
                'ct',
                $this->dbPrefix . 'customer_message',
                'cm',
                'ct.id_customer_thread = cm.id_customer_thread AND cm.id_employee > 0'
            )
            ->leftJoin(
                'cm',
                $this->dbPrefix . 'employee',
                'e',
                'cm.id_employee = e.id_employee'
            )
```

Also, one thing that worries me, you're supposed to get a list of customer threads here, but since you perform a join on messages you can get multiple rows returned for the same thread. Did you test what happens to the list when you start having a discussion in the thread? (maybe I'm overthinking and for some reason, it works)

But I believe you need to add a `groupBy(ct.id_customer_thread)` to your query no? And maybe the an `addOrderBy(cm.date_upd, 'DESC')` in order to get the latest message